omf-python/tests/test_grid_surface.py

```diff
@@ -0,0 +1,44 @@
+import omf_python
+from os import path
+from unittest import TestCase
+
+
+class TestGridSurface(TestCase):
+    def setUp(self) -> None:
+        self.omf_dir = path.join(path.dirname(__file__), "data")
+        self.one_of_everything = path.join(self.omf_dir, "one_of_everything.omf")
+
+    def test_should_contain_surface_geometry(self) -> None:
+        reader = omf_python.Reader(self.one_of_everything)
+        project, _ = reader.project()
+        grid_surface = project.elements()[3].geometry()
+        self.assertIsInstance(grid_surface, omf_python.GridSurface)
+
+        orientation = grid_surface.orient
+
+        self.assertEqual([-1.5, -1.5, 0], orientation.origin)
+        self.assertEqual([1, 0, 0], orientation.u)
+        self.assertEqual([0, 1, 0], orientation.v)
+
+        height_array = grid_surface.heights
+        self.assertEqual(9, height_array.item_count())
+        self.assertEqual(
+            [0, 0, 0, 0, 2, 0, 0, 0, 0], reader.array_scalars(height_array)
+        )
+
+        self.assertIsInstance(grid_surface.grid, omf_python.Grid2.Tensor)
+
+        self.assertEqual([2, 2], grid_surface.grid.count())
+
+        grid = grid_surface.grid.grid
```

Comment:
grid.grid is a bit odd here. Maybe we don't need `PyGrid2` and just have grid_surface.grid return either a PyGrid2Regular or PyGrid2Tensor ?

Reply:
That would probably be cleaner than using an enum. pyo3 doesn’t support complex enums yet, hence the awkward `.grid.grid`.

Reply:
Also my guess is the enum having methods might be causing the pyo3-stub-gen error